jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/execution/sam/PythonSamRunConfigurationIntegrationTest.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.execution.executors.DefaultDebugExecutor
+import com.intellij.testFramework.runInEdtAndWait
+import com.intellij.xdebugger.XDebuggerUtil
+import com.jetbrains.python.psi.PyFile
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.settings.SamSettings
+import software.aws.toolkits.jetbrains.utils.rules.PythonCodeInsightTestFixtureRule
+
+@RunWith(Parameterized::class)
+class PythonSamRunConfigurationIntegrationTest(private val runtime: Runtime) {
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun data(): Collection<Array<Runtime>> {
+            return listOf(
+                arrayOf(Runtime.PYTHON2_7),
+                arrayOf(Runtime.PYTHON3_6)
+            )
+        }
+    }
+
+    @Rule
+    @JvmField
+    val projectRule = PythonCodeInsightTestFixtureRule()
+
+    @Before
+    fun setUp() {
+        SamSettings.getInstance().executablePath = System.getenv().getOrDefault("SAM_CLI_EXEC", "sam")
+
+        val fixture = projectRule.fixture
+        fixture.addFileToProject(
+            "hello_world/__init__.py",
+            ""
+        )
+
+        val psiClass = fixture.addFileToProject(
+            "hello_world/app.py",
+            """
+            def lambda_handler(event, context):
+                return "Hello world"
+            """.trimIndent()
+        )
+
+        runInEdtAndWait {
+            fixture.openFileInEditor(psiClass.containingFile.virtualFile)
+        }
+    }
+
+    @Test
+    fun samIsExecuted() {
+        val runConfiguration = runConfiguration()
+        assertThat(runConfiguration).isNotNull
+
+        val executeLambda = executeLambda(runConfiguration)
+        assertThat(executeLambda.exitCode).isEqualTo(0)
+        assertThat(executeLambda.stdout).contains("Hello world")
+    }
+
+    @Test
+    @Ignore("Fails to attach debugger under CodeBuild")
```

Comment:
Might want to change this to an `assume` instead so that we can run it locally but that it gets skipped on codebuild.

Reply:
TIL. Can assume CI envVar is set, right?